Origin: commit:b728c9f54945203b8cce491ae0745b9229b7fb34
Author: Radek Krejci <rkrejci@cesnet.cz>
Subject: extensions BUGFIX reflect change from commit 01cdea196
Last-Update: 2018-12-06

The content member is not zero sized, so its 1 byte must be reflected
when initializing its content.

diff --git a/src/resolve.c b/src/resolve.c
index 788964f6bf08..b7995eb56bbd 100644
--- a/src/resolve.c
+++ b/src/resolve.c
@@ -5040,7 +5040,8 @@ resolve_extension(struct unres_ext *info, struct lys_ext_instance **ext, struct
         case LYEXT_COMPLEX:
             tmp_ext = realloc(*ext, ((struct lyext_plugin_complex*)e->plugin)->instance_size);
             LY_CHECK_ERR_GOTO(!tmp_ext, LOGMEM(ctx), error);
-            memset((char *)tmp_ext + sizeof **ext, 0, ((struct lyext_plugin_complex*)e->plugin)->instance_size - sizeof **ext);
+            memset((char *)tmp_ext + offsetof(struct lys_ext_instance_complex, content), 0,
+                   ((struct lyext_plugin_complex*)e->plugin)->instance_size - offsetof(struct lys_ext_instance_complex, content));
             (*ext) = tmp_ext;
             ((struct lys_ext_instance_complex*)(*ext))->substmt = ((struct lyext_plugin_complex*)e->plugin)->substmt;
             if (info->data.yang) {
-- 
2.19.2

